File: behaviorial/mediator/python/concrete_mediator.py
from component import Button, Checkbox, Component, TextField
from mediator import Mediator


class DialogMediator(Mediator):
    def __init__(self, checkbox: Checkbox, text_field: TextField, submit_button: Button):
        self.checkbox = checkbox
        self.checkbox.set_mediator(self)
        self.text_field = text_field
        self.text_field.set_mediator(self)
        self.submit_button = submit_button
        self.submit_button.set_mediator(self)

    def notify(self, sender: Component, event: str):
        if sender == self.checkbox and event == 'checked':
            self.text_field.set_visible(self.checkbox.is_checked())
        if sender == self.submit_button and event == 'click':
            self.submit_form()

    def submit_form(self):
        print("Form Submitted")
        print("Checkbox Checked:", self.checkbox.is_checked())
        if self.checkbox.is_checked():
            print("Text Field Value:", self.text_field.get_value())

File: behaviorial/mediator/typescript/concrete-mediator.ts
import { Button, Checkbox, Component, TextField } from './component';
import { Mediator } from './mediator';

export class DialogMediator implements Mediator {
	private checkbox: Checkbox;
	private textField: TextField;
	private submitButton: Button;

	constructor(checkbox: Checkbox, textField: TextField, submitButton: Button) {
		this.checkbox = checkbox;
		this.checkbox.setMediator(this);
		this.textField = textField;
		this.textField.setMediator(this);
		this.submitButton = submitButton;
		this.submitButton.setMediator(this);
	}

	notify(sender: Component, event: string): void {
		if (sender === this.checkbox && event === 'checked') {
			this.textField.setVisible(this.checkbox.isChecked());
		}
		if (sender === this.submitButton && event === 'click') {
			this.submitForm();
		}
	}

	private submitForm(): void {
		console.log('Form Submitted');
		console.log('Checkbox Checked:', this.checkbox.isChecked());
		if (this.checkbox.isChecked()) {
			console.log('Text Field Value:', this.textField.getValue());
		}
	}
}
